### PR TITLE
Discovery 3.0.0 w/ API docs build time

### PR DESCRIPTION
**Pull request by lzap**
Scratch builds to follow in a min

### CODE REVIEW

**Comment by lzap** [comment]
Got

```
cp cache/* /usr/share/foreman/vendor/cache/
cp foreman_bootdisk.rb /usr/share/foreman/bundler.d/
cp: cannot stat `foreman_bootdisk.rb': No such file or directory
make: *** [build] Error 1
```

http://ci.theforeman.org/job/packaging_build_deb_plugin/label=debian/423/console

Investigating...

**Comment by lzap** [comment]
Oh :-)

**Comment by domcleal** [comment]
> Oh :-)

Indeed :-)

File: plugins/ruby-foreman-discovery/debian/install [hunk 1]
```diff
@@ -1,2 +1,3 @@
 foreman_discovery.rb usr/share/foreman/bundler.d
 cache                usr/share/foreman/vendor
+apipie-cache/foreman_bootdisk    var/lib/foreman/public/apipie-cache/plugin
```
Comment (domcleal) [inline plugins/ruby-foreman-discovery/debian/install]: here too

**Comment by lzap** [comment]
Do I miss something in my patch for the 3.0.0 gem upgrade?

```
cp cache/* /usr/share/foreman/vendor/cache/
cp foreman_discovery.rb /usr/share/foreman/bundler.d/
cd /usr/share/foreman && ( \
		bundle install --local && \
		bundle exec rake db:migrate RAILS_ENV=development && \
		bundle exec rake plugin:apipie:cache[foreman_discovery] cache_part=resources \
		OUT=/var/lib/foreman/public/apipie-cache/plugin/foreman_discovery RAILS_ENV=development \
		)
[33mSome gems seem to be missing from your vendor/cache directory.[0m
[31mCould not find gem 'foreman_discovery (= 3.0.0) ruby' in the gems available on this machine.[0m
make: *** [build] Error 7
```

**Comment by domcleal** [comment]
Yeah, update plugins/ruby-foreman-discovery/debian/gem.list as that's the list of .gem files that the packages include from rubygems.org.

**Comment by lzap** [comment]
Do i need to add the deface to the gem.list as well?

```
31mCould not find gem 'deface (< 1.0) ruby', which is required by gem 'foreman_discovery (= 3.0.0) ruby', in any of the sources
```

**Comment by lzap** [comment]
Strange deface dep has not been added between 2.0 and 3.0...

**Comment by domcleal** [comment]
Ah, add ruby-foreman-deface to the build dependencies in the control file (similar to ruby-foreman-salt).

**Comment by lzap** [comment]
http://ci.theforeman.org/job/packaging_build_deb_plugin/427/

**Comment by domcleal** [comment]
Merged as 35d99c4 and also to deb/1.8, thanks @lzap.

I also added a changelog entry and set the min Foreman version to 1.8.0 in the control file.

**Comment by lzap** [comment]
Thanks, let me know once 1.8 is built so I can poke @ehelms and @stbenjam

**Comment by domcleal** [comment]
Monitor http://ci.theforeman.org/job/packaging_build_deb_plugin/429/, should be done in a few mins.

**Pull request closed without merging**